src/Compilers/CSharp/Portable/Binder/Semantics/AccessCheck.cs

```diff
@@ -243,6 +243,14 @@ private static bool IsNamedTypeAccessible(NamedTypeSymbol type, Symbol within, r
                 }
             }
 
+            // Check if the type has the EmbeddedAttribute and is in a different assembly.
+            // The EmbeddedAttribute marks a type as only visible within its own assembly.
+            var withinAssembly = within is AssemblySymbol assembly ? assembly : ((NamedTypeSymbol)within).ContainingAssembly;
+            if ((object)type.ContainingAssembly != withinAssembly && type.IsHiddenByCodeAnalysisEmbeddedAttribute())
```

Comment:
>(object)type.ContainingAssembly != withinAssembly [](http://example.com/codeflow?start=16&length=49)

This check looks incorrect, it is going to allow access to a type in a different added module.